### PR TITLE
Add `police_transfer` to the move types

### PR DESCRIPTION
**Pull request by alexdesi**
### Jira link

https://dsdmoj.atlassian.net/browse/P4-1781

### What?

This PR adds `police_transfer` to the allowed move_type

### Why?

because we want to support transfer police -> police locations

### Have you? (optional)

- [x] Updated API docs

### CODE REVIEW

File: app/models/move.rb [hunk 1]
```diff
@@ -150,22 +151,36 @@ def set_reference
   def set_move_type
     return if move_type.present?
```
Comment (martyn-w) [inline app/models/move.rb:153]: I'd add a comment to note that the order of if statements is important here

**Comment by martyn-w** [comment]
Don't forget to update `swagger/v1/move.yaml` and `swagger/v2/move.yaml`

File: app/models/move.rb [hunk 2]
```diff
@@ -150,22 +151,36 @@ def set_reference
...
-      else
-        'prison_transfer'
-      end
+    self.move_type = if is_a_prison_recall?
+                       'prison_recall'
+                     elsif is_a_court_appearance?
+                       'court_appearance'
+                     elsif is_a_police_tranfer?
+                       'police_transfer'
+                     elsif is_a_prison_transfer?
```
Comment (willfish) [inline app/models/move.rb:160]: We're missing a default

Comment (willfish) [inline app/models/move.rb:160]:
```suggestion
                     else
```

File: app/models/move.rb [hunk 3]
```diff
@@ -150,22 +151,36 @@ def set_reference
...
 
-    self.move_type =
-      if to_location.nil?
-        'prison_recall'
-      elsif to_location_is_court?
-        'court_appearance'
-      else
-        'prison_transfer'
-      end
+    self.move_type = if is_a_prison_recall?
```
Comment (willfish) [inline app/models/move.rb:154]:
```suggestion
    # TODO: The order is not important, here. Remove this from the model when we migrate to mandatory move_type under v2
    self.move_type = if is_a_prison_recall?
```

File: app/models/move.rb [hunk 4]
```diff
@@ -150,22 +151,36 @@ def set_reference
...
   end
 
   def ensure_event_nomis_ids_uniqueness
     nomis_event_ids.uniq!
   end
 
-  def to_location_is_court?
-    to_location&.location_type == 'court'
+  def is_a_police_tranfer?
+    to_location&.location_type == Location::LOCATION_TYPE_POLICE and
```
Comment (willfish) [inline app/models/move.rb:170]:
```suggestion
    to_location.police? && from_location.police?
```

File: app/models/move.rb [hunk 5]
```diff
@@ -150,22 +151,36 @@ def set_reference
...
+      (from_location&.location_type == Location::LOCATION_TYPE_POLICE)
+  end
+
+  def is_a_prison_recall?
+    to_location.nil?
+  end
+
+  def is_a_court_appearance?
+    to_location&.location_type == Location::LOCATION_TYPE_COURT
```
Comment (willfish) [inline app/models/move.rb:179]:
```suggestion
    to_location&.court?
```

File: app/models/move.rb [hunk 6]
```diff
@@ -150,22 +151,36 @@ def set_reference
...
+  end
+
+  def is_a_prison_transfer?
+    to_location&.location_type == Location::LOCATION_TYPE_PRISON
```
Comment (willfish) [inline app/models/move.rb:183]:
```suggestion
    to_location&.prison?
```

File: spec/models/move_spec.rb [hunk 7]
```diff
@@ -255,29 +255,38 @@
...
       it 'sets the move type to `prison_transfer` at validation time' do
         expect(move.move_type).to eq 'prison_transfer'
       end
     end
+
+    context 'when both to_location and from_location are police locations' do
+      let(:to_location) { build :location, :police }
+      let(:from_location) { build :location, :police }
+
+      it 'set move_type to police_transfer' do
```
Comment (willfish) [inline spec/models/move_spec.rb:286]:
```suggestion
      it 'sets move_type to police_transfer' do
```

Comment (willfish) [inline spec/models/move_spec.rb:286]:
```suggestion
      it 'sets move_type to `police_transfer`' do
```

**Comment by willfish** [comment]
Still missing the bits of swagger docs on the move, but very close to 🚀

**Comment by willfish** [comment]
```
 Running tests in:
 - Chrome 83.0.4103.116 / macOS 10.15.5

 Cancel allocation
 ✓ Cancel allocation

 New PMU allocation
 ✓ Create allocation and verify the result
 ✓ Check validation errors on allocation details page

 New move from Police Custody to Court
 ✓ With unfound person
 ✓ With existing person

 New move from Police Custody to Prison (recall)
 ✓ With a new person

 Cancel move from Police Custody
 ✓ Reason - `Made in error`
 ✓ Reason - `Supplier declined to move this person`
 ✓ Reason - `Another reason`

 New move from Prison to Court
 ✓ With unfound person
 ✓ With existing person

 New move from Prison to Prison
 ✓ With existing person

 Download moves
 ✓ As Police user
 ✓ As Prison user
 ✓ As STC user
 ✓ As Supplier user

 Smoke tests
 ✓ As Police user
 ✓ As STC user
 ✓ As Prison user
 ✓ As Supplier user
 ✓ As OCA user
```

**Pull request merged**